### PR TITLE
Switch to Jazzy for Harmonic

### PR DESCRIPTION
**Pull request by Amronos**
# 🦟 Bug fix

## Summary
Switches Rolling references in various places to Jazzy for Harmonic.
To be merged after https://github.com/gazebosim/ros_gz/pull/607 has been merged (and after Ionic is released?).

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [x] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: harmonic/ros2_overview.md [hunk 1]
```diff
@@ -26,9 +26,6 @@ Please follow the [Install Gazebo and ROS document](ros_installation)
 before starting this tutorial. A working installation of ROS 2 and Gazebo is
 required to go further.
 
-Important: Most of this functionality is only available in ROS 2 Rolling.
-We'll backport it to ROS 2 Jazzy soon.
-
```
Comment (azeey) [inline harmonic/ros2_overview.md:31]: Let's keep this until we actually make a Jazzy release.

File: harmonic/ros2_spawn_model.md [hunk 2]
```diff
@@ -5,7 +5,6 @@ Additionally, it's possible to spawn new models at any time. To do so using ROS
 we have provided the following mechanisms:
 
 ## Spawn a model using the launch file included in `ros_gz_sim`.
-*Currently only available in Rolling.*
```
Comment (azeey) [inline harmonic/ros2_spawn_model.md:8]: Let's keep this until we actually make a Jazzy release.

**Review by azeey — CHANGES_REQUESTED** [review]
Thanks for all the changes! There are some additional changes we're going to be making in ros_gz, so I don't think we'll be making a Jazzy release until then, so I suggest we keep the statements that say that the feature is only available in Rolling.

cc @caguero

File: harmonic/ros2_overview.md (hunk 1 above)
Comment (Amronos) [reply harmonic/ros2_overview.md]: Done in commit 0dcd5df.

**Review by azeey — APPROVED** [review]

**Pull request merged**